src/algebra/order/floor.lean

```diff
@@ -634,6 +645,12 @@ eq_of_forall_ge_iff (λ z, by rw [neg_le, ceil_le, le_floor, int.cast_neg, neg_l
 
 lemma lt_ceil : z < ⌈a⌉ ↔ (z : α) < a := lt_iff_lt_of_le_iff_le ceil_le
 
+@[simp] lemma one_le_ceil_iff : 1 ≤ ⌈a⌉ ↔ 0 < a :=
+begin
+  have : 0 < a ↔ ((0 : ℤ) : α) < a, { norm_cast, },
+  rw [this, ← lt_ceil, lt_iff_add_one_le, zero_add],
+end
+
```

Comment:
Can you prove the more general lemma about `int.cast` first? Same for `floor_le_neg_one_iff`.

Reply:
I'm afraid I don't understand what you mean here. Could you spell it out?

Reply:
```
@[simp] lemma floor_le_sub_one_iff : ⌊a⌋ - 1 ≤ n ↔ a < n := sorry
@[simp] lemma add_one_le_ceil_iff : n + 1 ≤ ⌈a⌉ ↔ n < a :=
```

Reply:
Thanks, now done.